appcues/src/test/java/com/appcues/action/ActionProcessorTest.kt

```diff
@@ -153,8 +157,10 @@ internal class ActionProcessorTest : KoinTest {
                                 experienceRenderer = get(),
                             )
                         }
-                        scoped { StateMachine(get(), get(), get(), initState) }
+                        factory { StateMachine(get(), get(), get(), initState) }
```

Comment:
changing here from scoped to factory makes so whenever we "get" its a new instance.